prediction_market_agent_tooling/tools/cow/cow_manager.py

```
@@ -0,0 +1,111 @@
+            return asyncio.run(
+                get_order_quote(
+                    order_quote_request=order_quote_request,
+                    order_side=order_side,
+                    order_book_api=self.order_book_api,
+                )
+            )
+
+        except UnexpectedResponseError as e1:
+            logger.error(f"Unexpected response error: {e1.message}")
+            if "NoLiquidity" in e1.message:
+                raise NoLiquidityAvailableOnCowException(e1.message)
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)